### ISSUE TITLE
Per-instance lvm stripping configuration

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#144]
Originally reported of Google Code with ID 88.

```
Hi,

it could be interesting to let ganeti support lvm stripping. Because it is 
a feature lv-related, it can be done on a vm-basis (with maybe cluster 
default as for most of the ganeti parameters).

Looking at http://git.ganeti.org/?
p=ganeti.git;a=blob;f=lib/bdev.py;hb=HEAD#l395 it seems not to be done yet. 

It is a scheduled improvement or is it something you don't implemented yet 
because of some good (technical) reason I am not aware of ? 

Thanks again for your work !

Regards,

Maxence
```
Originally added on 2010-02-26 08:50:04 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
It is implemented, just not on a per-VM basis, but cluster wide. The exact lines
you're looking at implement it.

Make sure to configure ganeti with --with-lvm-stripecount=N.

regards,
iustin
```
Originally added on 2010-02-26 08:54:23 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2010-02-26 08:54:59 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
ehhh, sorry for that ... should read carefully. Anyway,I don't want to use lvm 
stripping for most of my vms. Is there a way to override the parameter on a per-vm 
basis ? 
```
Originally added on 2010-02-26 08:58:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
No, not currently, which is why I renamed the bug :)

iustin
```
Originally added on 2010-02-26 09:06:54 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
:) Just have to wait for it then :D Thanks for all !

Maxence
```
Originally added on 2010-02-26 09:09:26 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
hmmm, I'm thinking of something. The lvm strip count is in _autoconf.py, overridden 
by constants.py . It is used then used in bdev. 

I guess I can change the lvm strip count in constants.py for my 2 new nodes, so that 
my first 10 nodes just uses strips=1, and my 2 new nodes will uses strip = 3.

Can this trigger any issue ?

Regards,

Maxence
```
Originally added on 2010-03-06 08:54:00 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
No, that will be fine. The only thing it can do is that if an instance is living
primary on old nodes, secondary on new nodes (or viceversa), then one side will be
slower (no stripping). But that is just a performance issue.

iustin
```
Originally added on 2010-03-06 10:34:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2010-03-12 19:14:31 +0000 UTC.

Added to Milestone: Release2.2

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unfortunately this didn't make it in 2.2. Re-targetting at 2.3.

iustin
```
Originally added on 2010-07-28 23:17:12 +0000 UTC.

Changed State: Accepted
Added to Milestone: Release2.3
Removed from Milestone: Release2.2

**Comment by Ganeti-Issues-Migrator** [comment]
```
Apparently this did not make it to 2.3 either?
```
Originally added on 2010-12-06 22:18:50 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Unfortunately not, re-targetting once again.
```
Originally added on 2010-12-07 14:19:48 +0000 UTC.

Added to Milestone: Release2.4
Removed from Milestone: Release2.3

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2011-02-18 13:41:20 +0000 UTC.

Added to Milestone: Unplanned
Removed from Milestone: Release2.4

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2011-08-05 14:45:42 +0000 UTC.

Removed from Milestone: Unplanned

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been implemented and will be appear in the 2.6 release.
```
Originally added on 2012-05-22 16:10:03 +0000 UTC.

Changed State: Started
Added to Milestone: Release2.6

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking as fixed, since 2.6 will be released soon now (release candidates are available for testing).
```
Originally added on 2012-07-19 11:05:30 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:07 +0000 UTC.

Changed State: Released
